File: src/utils/validate.js
module.exports.number = (variable, variable_name) =>{

    if(typeof variable != "number"){
        throw "Invalid data. "+variable_name+ " must be a number";
    }
}

module.exports.string = (variable, variable_name) =>{

    if(typeof variable != "string"){
        throw "Invalid data. "+variable_name+ " must be a string";
    }
}

module.exports.datetimeString = (variable, variable_name) => {
    if(typeof variable != "string"){
        throw "Invalid data. "+variable_name+ " must be a datetime string";
    }

    if(new Date(variable) == "Invalid Date"){
        throw "Invalid data. "+variable_name+ " must be a valid datetime string";
    }
}

File: src/utils/validate.py
import datetime

def number(variable, variable_name):
    if(isinstance(variable, int) != True):
        raise Exception("Invalid data. " + variable_name + " must be a number")

def string(variable, variable_name):
    if(isinstance(variable, str) != True):
        raise Exception("Invalid data. " + variable_name + " must be a string")

def datetimeString(variable, variable_name):
    if(isinstance(variable, str) != True):
        raise Exception("Invalid data. " + variable_name + " must be a datetime string")

    if(isinstance(variable, datetime.datetime) != True):
        raise Exception("Invalid data. " + variable_name + " must be a datetime string")
